src/test/test_utils.rs

```diff
@@ -538,3 +557,23 @@ pub fn verify_syscall_result(
     }
     Ok(())
 }
+
+/// Run a function and check that it returns within an expected duration.
+pub fn check_fn_exec_duration<F>(
+    expected: Duration,
+    tolerance: Duration,
+    f: F,
+) -> anyhow::Result<()>
+where
+    F: FnOnce() -> anyhow::Result<()>,
+{
```

Comment:
You can make this function a bit more flexible using:

```rust
pub fn check_fn_exec_duration<F, E>(
    expected: Duration,
    tolerance: Duration,
    f: F,
) -> anyhow::Result<()>
where
    F: FnOnce() -> Result<(), E>,
    anyhow::Error: From<E>,
{
```

This way `F` can return almost any error (including those from `result_assert()`), not just anyhow errors.